Source/WebCore/Modules/indexeddb/server/SQLiteMemoryIDBBackingStore.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <WebCore/SQLiteIDBBackingStore.h>
+
+namespace WebCore {
+namespace IDBServer {
+
+// SQLite-backed in-memory IndexedDB backing store.
+// Unlike MemoryIDBBackingStore (which uses HashMaps), this uses SQLite's
+// in-memory database (":memory:") for better scalability and ACID guarantees.
+// Unlike SQLiteIDBBackingStore (which persists to disk), all data is ephemeral
+// and lost when the backing store is destroyed.
+class SQLiteMemoryIDBBackingStore final : public SQLiteIDBBackingStore {
```

Comment:
Why do we need a new `SQLiteMemoryIDBBackingStore` class? Can we just create a `SQLiteIDBBackingStore` and update its functions to do different things based on whether `SQLiteIDBBackingStore::m_databaseDirectory` is null?
(In the current implementation `SQLiteMemoryIDBBackingStore` inherits `SQLiteIDBBackingStore`, but `SQLiteIDBBackingStore` is persistent and `SQLiteMemoryIDBBackingStore` is not)